### PR TITLE
Added DAG support

### PR DESCRIPTION
**Pull request by meyer1994**
# Changelog
- Added the `dag` attribute to client
- The `dag` attribute has the following methods:
  - `put`
  - `get`
  - `resolve`
  - `import`
  - `export`
- Added tests
- Added docs

## Notes
This implementation was basically copying and pasting from other parts of the codebase. The tests were created in the same fashion. :sweat_smile:

### CODE REVIEW

**Comment by meyer1994** [comment]
About the error in Travis, it seems that version 0.4.22 does not support `ipfs dag import/export`. By looking at the changelog, it seems that support for it was added in [v.0.5.0][1]

[1]: https://github.com/ipfs/go-ipfs/releases/tag/v0.5.0

File: ipfshttpclient/client/dag.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+		-------
+			dict
+				Cid with the address of the dag object
+		"""
+		args = (str(cid),)
+		return self._client.request('/dag/resolve', args, decoder='json', **kwargs)
+
+	@base.returns_single_item(base.ResponseBase)
+	def imprt(self, data: ty.IO, **kwargs: base.CommonArgs):
+		"""Imports a .car file with a DAG to IPFS
```
Comment (ntninja) [inline ipfshttpclient/client/dag.py:104]: This should include a comment about why the name is spelled incorrectly (it wasn't immediately obvious to me and I'm sure there are lots of less experienced Python programmers that wouldn't get it all and just be annoyed by what they think is a typo).

File: ipfshttpclient/client/dag.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+from . import base
+
+from .. import multipart
+
+
+class Section(base.SectionBase):
+	@base.returns_single_item(base.ResponseBase)
+	def get(self, cid: str, **kwargs: base.CommonArgs):
+		"""Get and serialize the DAG node named by CID.
```
Comment (ntninja) [inline ipfshttpclient/client/dag.py:11]: Nit: Tone of voice is different here from other other doc-strings.

**Comment by ntninja** [comment]
Merged in #232. Thank you!

**Comment by ntninja** [comment]
@meyer1994: I just realized that this was missing support for the optional parameters of [/dag/put](https://ipfs.io/ipns/docs.ipfs.io/reference/http/api/#api-v0-dag-put). Could you add those?

It would allow importing directories based on lists of CIDs like requested in #220. :slightly_smiling_face:

**Comment by meyer1994** [comment]
@ntninja I will try to do it this weekend :)

**Comment by ntninja** [comment]
Thanks! :slightly_smiling_face:

**Comment by meyer1994** [comment]
Created PR #238

**Pull request closed without merging**